shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/compiler/SampleNodeFixedWithCondition.java

```diff
@@ -75,6 +75,16 @@ public void collectFreeVariables(Set<String> names) {
 		}
 	}
 
+	@Override
+	public Consumption consumptionCheck(Consumption previous, Consumer<String> errorHandler) {
+		if (previous == Consumption.VARIABLE) {
+			errorHandler.accept(String.format("%d:%d: Cannot consume a fixed number of items after a variable number.",
+					limitExpression.line(), limitExpression.column()));
+			return Consumption.BAD;
+		}
+		return previous;
```

Comment:
Why does this `consumptionCheck` (and the one in SampleNodeFixed) return `previous`, while the one in SampleNodeSquish returns `Consumption.VARIABLE`? (aka why wouldn't the Fixed ones return `Consumption.FIXED`?)

Reply:
I'm in favour of explicit switch blocks.